geth/jail/jail.go

```diff
@@ -137,47 +140,68 @@ func (j *Jail) initCell(cell *Cell) error {
 }
 
 // CreateAndInitCell creates and initializes a new Cell.
-func (j *Jail) createAndInitCell(chatID string, code ...string) (*Cell, error) {
+func (j *Jail) createAndInitCell(chatID string, code ...string) (*Cell, string, error) {
 	cell, err := j.obtainCell(chatID, false)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	if err := j.initCell(cell); err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
-	// Run custom user code
+	resultString, err := j.runJSCode(cell, code)
+	if err != nil {
+		return nil, "", err
+	}
+
+	return cell, resultString, nil
+}
+
+// runJSCode runs the provided code in the provided cell and
+// returns results of the last statement.
+func (j *Jail) runJSCode(cell *Cell, code []string) (string, error) {
+	if cell == nil {
+		return "", errors.New("no cell provided")
+	}
+
+	var resultRef *otto.Value
+
 	for _, js := range code {
-		_, err := cell.Run(js)
+		codeResult, err := cell.Run(js)
 		if err != nil {
-			return nil, err
+			return "", err
 		}
+		resultRef = &codeResult
 	}
 
-	return cell, nil
+	resultString := ResponseCellCreatedEmpty
+	if resultRef != nil {
```

Comment:
So the result will be only from the last JS code string?

Reply:
Yes. And the Closure counterpart is already tweaked to support this behaviour: [status-react#2587](https://github.com/status-im/status-react/pull/2587)